### PR TITLE
Debt/met 3315 move harvesting code to metis

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   * Constructor.
+   *
+   * @param repositoryUrl The base url of the repository.
+   * @param metadataPrefix The metadata prefix (optional).
+   * @param setSpec The set spec value.
+   * @param from The from value (optional).
+   * @param until The until value (optional).
+   */
+  public OaiHarvest(String repositoryUrl, String metadataPrefix, String setSpec,
+          Instant from, Instant until) {
```
Comment (JoanaCMS) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvest.java:26]: Sugestion: Another good opportunity for Builder I believe

File: metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    CompressedFileExtractor.extractFile(Path.of(DESTINATION_DIR + FILE_NAME + ".tar.gz"),
+            Path.of(DESTINATION_DIR));
+    Collection<File> files = getXMLFiles(DESTINATION_DIR + FILE_NAME);
+    assertNotNull(files);
+    assertEquals(XML_FILES_COUNT, files.size());
+  }
+
+  @Test
+  public void shouldUnpackTheTarGzFilesRecursivelyWithCompressedXMLFiles() throws IOException {
+    CompressedFileExtractor.extractFile(Path.of(DESTINATION_DIR + FILE_NAME2 + ".tar.gz"),
```
Comment (JoanaCMS) [inline metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java:34]: The point of this test is to unpack the TarGz files with compressed XML files. However, `FILE_NAME2` has a value String of `gzFileWithCompressedGZFiles`. 
This is is a bit misleading for me, do you want to test this with compressed XML files or you want to get XML files from compressed Gz files?

Comment (jochen-vermeulen) [reply metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java:34]: No, the idea here is to test it with nested compressed files. So this is a gz file that itself contains other gz files inside it. Does that make sense?

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvest.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvest.java:26]: Yes, this one I agree with.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/CloseableHttpOaiClient.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+   * @param baseUrl The base URL of the OAI-PMH repository.
+   * @param userAgent The user agent we wish to set for connections.
+   * @param numberOfRetries The number of retries that we apply to connections.
+   * @param timeBetweenRetries The time (in ms) between any connection retry.
+   * @param requestTimeout The request timeout (in ms) for connections.
+   * @param connectionTimeout The connection timeout (in ms) for connections.
+   * @param socketTimeout The socket timeout (in ms) for connections.
+   */
+  public CloseableHttpOaiClient(String baseUrl, String userAgent, int numberOfRetries,
+          int timeBetweenRetries, int requestTimeout, int connectionTimeout, int socketTimeout) {
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/CloseableHttpOaiClient.java:84]: For this one, I have a different approach to suggest. I created a separate connection settings class, with setters. Seems like a better approach.

File: metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java (hunk 2 above)
Comment (JoanaCMS) [reply metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java:34]: Yes, now I understand, thanks!

**Pull request merged**